crossgovernmentservices/ags_gateway
# Use fake OIDC server and client for clearer tests

## What

Refactored oidc_client tests to use a reusable mock OIDC server, to remove tight coupling between tests and implementation and test at a higher level of abstraction

## How to review

1. Run tests with `./run-tests`
2. Read test code in `tests/integration/test_oidc_client.py` and `tests/integration/test_oidc_provider.py`
3. See that tests focus on high-level HTTP requests flow in OIDC authentication process, without implementation detail obscuring it
